src/Stack/Build/Execute.hs

```diff
@@ -584,14 +585,12 @@ executePlan' installedMap0 targets plan ee@ExecuteEnv {..} = do
         [] -> return ()
         ids -> do
             localDB <- packageDatabaseLocal
-            forM_ ids $ \(id', (ident, reason)) -> do
-                logInfo $
-                    fromString (packageIdentifierString ident) <>
-                    ": unregistering" <>
-                    if T.null reason
-                        then ""
-                        else " (" <> RIO.display reason <> ")"
-                unregisterGhcPkgId wc cv localDB id' ident
+            case cv of
+                ACGhc v | v >= mkVersion [8, 0, 1] ->
+                    for_ (chunksOf 500 ids) $ unregisterPackages wc localDB
+                _ ->
+                  for_ ids $ \(gid, (ident, reason)) -> do
+                      unregisterSinglePackage cv localDB gid ident reason
```

Comment:
We actually have 3 cases here:

-  GHC >= 8.0.1 uses `unregisterPackages`
-  GHC < 8.0.1 && >= 7.9 uses `unregisterSinglePackage` with a `GhcPkgId`
-  GHC < 7.9 uses `unregisterSinglePackage` with an identifier.

It may be worth reflecting this here, to make clear that this is the case.
Or not — if it's better to focus on two cases: new vs old.

Reply:
Actually, if you do the `case` here, you don't have to pass the `ActualCompiler` which greatly simplifies the `unregister*` functions.